docs/configuration/OVERVIEW.md

```diff
@@ -156,106 +214,290 @@ export interface MyConfig extends Configuration {
   myBooleanField: false,
   myNumberField: 0,
-  myStringListField: ['firstDefaultValue', 'secondDefaultValue']
+  myUnionTypeField: 'before',
+  myReferencedUnionTypeField: 'top',
+  myStringListField: ['firstDefaultValue', 'secondDefaultValue'],
+  myNestedField: [
+    {
+      'myNestedStringField': 'myNestedStringField default value 1',
+      'myNestedBooleanField': false,
+      'myNestedNumberField': 15
+    },
+    {
+      'myNestedStringField': 'myNestedStringField default value 2',
+      'myNestedBooleanField': true,
+      'myNestedNumberField': 10
+    }
+  ]
 };
 ```
 
-### Component file (*.component.ts)
+> [!WARNING]
+> The default configuration const needs to be explicitly typed with the configuration interface and contain no variable references.
+
+> [!NOTE]
+> The order for Nested configuration does not matter.
+
+`UnionTypes` are supported in 2 cases:
+
+1) inline definition (see above `myUnionTypeField`)
+2) reference to a union type that is defined in the same configuration file (see above `myReferencedUnionTypeField` and `Position`).
+
+### Configuration tags
+
+To implement this feature, you should add the tags in the JSDoc of the configuration interface while respecting the correct format, for example:
+
+```typescript
+/**
+ * MyConfig description
+ * @tags [tag1, tag2, tag3]
+ */
+export interface MyConfig extends Configuration {
+
+}
+```
+
+> [!NOTE]
+> We also support the multiline format for the tags, for example:
+> ```typescript
+> /**
+>  * MyConfig description
+>  * @tags [
+>  *   tag1,
+>  *   tag2,
+>  *   tag3
+>  * ]
+>  */
+> ```
+
+These tags will be exported inside the extracted metadata (see the [CMS Adapters documentation](../cms-adapters)) provided
+they are supported in the [CMS JSON schema](https://github.com/AmadeusITGroup/otter/tree/main/packages/%40o3r/configuration/schemas/configuration.metadata.schema.json).
+Please refer to the schema for the latest supported model.
+
+For instance, if you want to add a title to your component's configuration as a way to have a user friendly naming and a label for your property, you can set the following JSTags:
+```typescript
+/**
+ * This is an incredible config but the name is not so easy to read for CMS users
+ *
+ * @title My Incredible Config
+ */
+export interface MyConfigWithADifficultName extends Configuration {
+  /**
+   * My great property
+   *
+   * @label Human readable title
+   */
+  myConfigProperty: string;
+}
+```
+
+If you use any non-supported tags in your tsdocs, they will be ignored by the extractor. For instance, in the following example, the `unsupportedTag` will not be part of the extracted metadata.
+
+```typescript
+/**
+ * Yet another Configuration
+ *
+ * @unsupportedTag This tag will be ignored by the configuration extractor
+ */
+export interface MyConfig extends Configuration {
+  /**
+   * Some description
+   */
+  someConfigProperty: string;
+}
+```
+
+### Configuration categories
```

Comment:
Maybe we can add that we expose a linter for that `docs/linter/eslint-plugin/rules/o3r-widget-tags.md`

Reply:
And a vscode extension for autocomplete

Reply:
I will open an issue once the PR is merged

Reply:
https://github.com/AmadeusITGroup/otter/issues/1864